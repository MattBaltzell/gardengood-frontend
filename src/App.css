.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: 2rem;
  background-color: #282c34;
  color: rgba(255, 255, 255, 0.6);
}

.container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  gap: 2rem;
  width: 100%;
  max-width: 1200px;
}

.Plant {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: left;
  justify-content: flex-start;
  width: 100%;
  text-align: center;
  /* background-color: #3d4553; */
  gap: 1rem;
  border-radius: 3px;
  /* box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2); */

  border-radius: 50%;
}

.Plant h3 {
  font-size: 1rem;
  transition: all 0.2s ease-in-out;
}

.Plant-img {
  object-fit: cover;
  width: 100%;
  border-radius: 50%;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
  position: relative;
  transform: translateY(0);
  transition: all 0.2s ease-in-out;
}

.Plant-img:hover {
  filter: brightness(1.1);
  transform: translateY(-0.5rem) scale(1.05);
  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.4);
}
.Plant:hover h3 {
  color: #fff;
  transform: translateY(-0.3rem) scale(1.04);
}
