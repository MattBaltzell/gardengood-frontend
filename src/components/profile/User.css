.User {
}

.User__info-header {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.User__info-header p {
  font-size: 2rem;
  font-weight: 200;
}
