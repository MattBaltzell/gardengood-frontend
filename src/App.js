import { useState, useEffect } from "react";
import "./App.css";
import axios from "axios";

import PlantGrid from "./components/Plant/PlantGrid";
function App() {
  const [plants, setPlants] = useState([]);

  useEffect(() => {
    async function getPlants() {
      try {
        const res = await axios({
          url: "http://localhost:3001/plants",
          method: "GET",
          headers: {
            Authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3RhZG1pbiIsImlzQWRtaW4iOmZhbHNlLCJpYXQiOjE2Nzc2NDA3Mzl9.HoNDrjYXseNCN6CIKkCB9FqT6ecJumAVNE6OeTg1WLk`,
          },
        });
        const plants = res.data.plants;

        setPlants(plants);
      } catch (error) {
        console.error(error);
      }
    }
    getPlants();
  }, []);
  return (
    <div className="App">
      <div className="container">
        <PlantGrid plants={plants} />
      </div>
    </div>
  );
}

export default App;
