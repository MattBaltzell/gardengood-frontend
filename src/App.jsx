import React, { useState, useEffect } from "react";
import "./App.css";
import Router from "./components/Navigation-Routes/Router";
import Navbar from "./components/Navigation-Routes/Navbar";
import GardenGoodApi from "./api/api";
import UserContext from "./auth/UserContext";
import jwt_decode from "jwt-decode";

function App() {
  const [currUser, setCurrUser] = useState(null);
  const [token, setToken] = useState(localStorage.getItem("gardengood-token"));
  const [userWasUpdated, setUserWasUpdated] = useState(false);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getCurrUser = async () => {
      if (token) {
        try {
          GardenGoodApi.token = token;
          let { username } = jwt_decode(token);
          const user = await GardenGoodApi.getCurrentUser(username);
          setCurrUser(user);
          localStorage.setItem("gardengood-token", token);
        } catch (error) {
          console.error(error);
        }
      }
    };
    getCurrUser();

    setUserWasUpdated(false);
  }, [token, userWasUpdated]);

  const handleIsLoading = (bool) => {
    setIsLoading(bool);
  };

  const login = async ({ username, password }) => {
    try {
      const token = await GardenGoodApi.login({ username, password });
      setToken(token);
    } catch (error) {
      return { message: error };
    }
  };

  const signup = async ({
    username,
    password,
    firstName,
    lastName,
    email,
    zipCode,
  }) => {
    try {
      const token = await GardenGoodApi.signup({
        username,
        password,
        firstName,
        lastName,
        email,
        zipCode,
      });
      setToken(token);
    } catch (error) {
      return { message: error };
    }
  };

  const logout = async () => {
    try {
      setToken(null);
      setCurrUser(null);
      GardenGoodApi.token = null;
      localStorage.setItem("gardengood-token", "");
    } catch (error) {
      return { message: error };
    }
  };

  return (
    <div className="App">
      <UserContext.Provider value={currUser}>
        <Navbar logout={logout} />
        <Router
          login={login}
          signup={signup}
          logout={logout}
          handleIsLoading={handleIsLoading}
          isLoading={isLoading}
        />
      </UserContext.Provider>
    </div>
  );
}

export default App;
