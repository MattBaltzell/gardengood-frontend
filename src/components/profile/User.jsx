import React, { useContext } from "react";
import { useParams, Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import NotFound from "../../NotFound";
import "./User.css";
import Unauth from "../../Unauth";

const User = () => {
  const { username } = useParams();
  const currUser = useContext(UserContext);

  if (!currUser.username) {
    return <NotFound />;
  }

  return currUser.username !== username ? (
    <Unauth />
  ) : (
    <>
      <main className="User">
        <div className="User__info-container">
          <h1>Account Info</h1>

          <div className="User__info-container-section">
            <div className="User__info-group">
              <h4>Username</h4>
              <p>{currUser.username}</p>
            </div>
            <div className="User__info-group">
              <h4>Email</h4>
              <p>{currUser.email}</p>
            </div>
            <div className="User__info-group">
              <h4>First Name</h4>
              <p>{currUser.firstName}</p>
            </div>
            <div className="User__info-group">
              <h4>Last Name</h4>
              <p>{currUser.lastName}</p>
            </div>
            <div className="User__info-group">
              <h4>Zip Code</h4>
              <p>{currUser.zipCode}</p>
            </div>
          </div>

          <Link to={`/users/${currUser.username}/update`}>
            Update account info
          </Link>
        </div>
      </main>
    </>
  );
};

export default User;
