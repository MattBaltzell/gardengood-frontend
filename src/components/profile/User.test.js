import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import User from "./User";
import App from "../../App";

import UserContext from "../auth/UserContext";

const user = {
  username: "testuser",
  email: "test@test.com",
  firstName: "firstname",
  lastName: "lastname",
  zipCode: "36830",
  joinAt: "2023-03-03 18:24:48.215773",
  isAdmin: false,
};

describe("User component tests", function () {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <User />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <User />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
