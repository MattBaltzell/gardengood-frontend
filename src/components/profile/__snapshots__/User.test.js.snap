// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component tests matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="Unauth"
  >
    <h1>
      Not so fast!
    </h1>
    <h3>
      You do not have permission to view that page!
    </h3>
    <a
      href="/"
    >
      Return to home page.
    </a>
  </main>
</DocumentFragment>
`;
